operator_ui/src/containers/Header.tsx

```diff
@@ -12,96 +17,54 @@ import Typography from '@material-ui/core/Typography'
-import AvatarMenu from 'components/AvatarMenu'
-import { submitSignOut } from 'actions'
-import fetchCountSelector from 'selectors/fetchCount'
 import ReactResizeDetector from 'react-resize-detector'
 import classNames from 'classnames'
+import LoadingBar from '../components/LoadingBar'
+import MainLogo from '../components/Logos/Main'
+import AvatarMenu from '../components/AvatarMenu'
+import { submitSignOut } from '../actions'
+import fetchCountSelector from '../selectors/fetchCount'
 
-const drawerWidth = 240
+const SHARED_NAV_ITEMS = [
+  ['/jobs', 'Jobs'],
+  ['/runs', 'Runs'],
+  ['/bridges', 'Bridges'],
+  ['/transactions', 'Transactions'],
+  ['/config', 'Configuration']
+]
 
-const styles = theme => {
-  return {
-    appBar: {
-      backgroundColor: theme.palette.common.white,
-      zIndex: theme.zIndex.modal + 1
-    },
-    toolbar: {
-      paddingLeft: theme.spacing.unit * 5,
-      paddingRight: theme.spacing.unit * 5
-    },
+const drawerStyles = ({ palette, spacing, zIndex }: Theme) =>
+  createStyles({
     menuitem: {
-      padding: theme.spacing.unit * 3,
+      padding: spacing.unit * 3,
       display: 'block'
     },
-    horizontalNav: {
-      paddingTop: 0,
-      paddingBottom: 0
-    },
-    horizontalNavItem: {
-      display: 'inline'
-    },
-    horizontalNavLink: {
-      color: theme.palette.secondary.main,
-      paddingTop: theme.spacing.unit * 3,
-      paddingBottom: theme.spacing.unit * 3,
-      textDecoration: 'none',
-      display: 'inline-block',
-      borderBottom: 'solid 1px',
-      borderBottomColor: theme.palette.common.white,
-      '&:hover': {
-        borderBottomColor: theme.palette.primary.main
-      }
-    },
-    activeNavLink: {
-      color: theme.palette.primary.main,
-      borderBottomColor: theme.palette.primary.main
-    },
     drawerPaper: {
-      backgroundColor: theme.palette.common.white,
-      paddingTop: theme.spacing.unit * 7,
+      backgroundColor: palette.common.white,
+      paddingTop: spacing.unit * 7,
       width: drawerWidth
     },
     drawerList: {
       padding: 0
     }
-  }
-}
-
-const SHARED_NAV_ITEMS = [
-  ['/jobs', 'Jobs'],
-  ['/bridges', 'Bridges'],
-  ['/transactions', 'Transactions'],
-  ['/config', 'Configuration']
-]
+  })
 
-class Header extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = { drawerOpen: false }
-  }
-
-  setDrawerOpen(isOpen) {
-    this.setState({
-      drawerOpen: isOpen
-    })
-  }
+interface IDrawerProps extends WithStyles<typeof drawerStyles> {
+  authenticated: boolean
+  drawerOpen: boolean
+  toggleDrawer: () => undefined
+}
 
-  render() {
-    const toggleDrawer = () => this.setDrawerOpen(!this.state.drawerOpen)
-    const signOut = () => this.props.submitSignOut()
-    const { classes, fetchCount, url } = this.props
-    const isNavActive = (current, to) => `${to && to.toLowerCase()}` === current
-    const drawer = (
-      <Drawer
+const Drawer = withStyles(drawerStyles)(
```

Comment:
I'm not able to sign out with a drawer in dev tools mobile view

Reply:
while at this, what do you think about extracting drawer to a component file? I feel like Header is getting too big and Drawer has reasons to exist as a component separately

Reply:
Fixed signout on mobile.

Adding the Typescript definitions does make the file a little large but I like the contextual grouping of these components in the same file. 

What advantage do you think breaking them into separate files brings?

Reply:
I'd personally group them into the same folder directory under `Header`, but this is just a preference, so, resolved